.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,157 @@
+
+  premerge:
+    runs-on: ubuntu-latest
+    # needs: get-variables # this can be uncommented after the reusable job works (see above)
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Don't save Bash session history
+        run: unset HISTFILE
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          check-latest: true
+
+      - name: Configure kernel (increase watchers)
+        run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+
+      # this step might be replaced with a reusable workflow get-variables.yml (see above)
+      - name: Get variables (branch name, commit hash, yarn cache directory)
+        id: get-variables
+        run: |
+          echo "::set-output name=shortref::${GITHUB_REF#refs/*/}";
+          echo "::set-output name=commitsha::$(echo ${GITHUB_SHA})";
+          echo "::set-output name=yarncachedir::$(yarn cache dir)";
+
+      - name: Workspace cache
+        uses: actions/cache@v3
+        id: workspace-cache
+        env:
+          cache-name: workspace-cache
+        with:
+          path: |
+            ~/.cache
+            ./node_modules
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ steps.get-variables.outputs.shortref }}-workspace-${{ steps.get-variables.outputs.commitsha }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ steps.get-variables.outputs.shortref }}-workspace-
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+            ${{ runner.os }}-
+          ## the above might be replaced with the commented code beneath if a reusable workflow get-variables.yml is used (see above)
+          # key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ needs.get-variables.outputs.shortref }}-workspace-${{ needs.get-variables.outputs.commitsha }}
+          # restore-keys: |
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ needs.get-variables.outputs.shortref }}-workspace-
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+          #   ${{ runner.os }}-
+
+      - name: Install project dependencies
+        run: yarn install --frozen-lockfile --non-interactive
```

Comment:
I would think that everything from this line and upwards would be a good candidate for a reusable workflow `setup-project.yml`.

Reply:
The comment was resolved during the call.